distributed/deploy/tests/test_local.py

```diff
@@ -831,3 +831,20 @@ def test_starts_up_sync(loop):
         assert len(cluster.scheduler.workers) == 2
     finally:
         cluster.close()
+
+
+def test_dont_select_closed_worker():
+    cluster = LocalCluster(n_workers=0, processes=False, threads_per_worker=2)
+    c = Client(cluster)
+    cluster.scale(2)
+    assert c == get_client()
+
+    c.close()
+    cluster.close()
+
+    cluster2 = LocalCluster(n_workers=0, processes=False, threads_per_worker=2)
```

Comment:
It looks like we don't close `cluster2`.  Could I ask you to include the `distributed.utils_test.clean` context manager within this test to make sure that it doesn't leak resources?  We'll probably also want a `cluster2.close` call at the end.  Otherwise there are a few workers around sending messages about that might interfere with other unrelated tests in the future.

Reply:
Woops, indeed. Fixed.